src/actions.js

```diff
@@ -75,76 +74,71 @@ export function addCheckResult(
 // ASYNC (THUNK) ACTIONS.
 
 // Fetching the statuses.
-export function requestStatus(version: ?string): ThunkAction<void> {
+export function requestStatus(version: ?string) {
   const notifyChanges = (checkTitle, status) => {
     if (Notification.permission === 'granted') {
       new Notification(`${checkTitle} : status changed (${status}).`);
     }
   };
 
-  return function(dispatch: Dispatch, getState: GetState) {
+  return async function(dispatch: Dispatch, getState: GetState) {
     const versionToCheck = version || getState().version;
     if (!versionToCheck) {
       return;
     }
     // Save previous results so we can check if something changed.
     const prevResults = getState().checkResults;
     dispatch(setVersion(versionToCheck));
-    getReleaseInfo(versionToCheck)
-      .then(releaseInfo => {
-        dispatch(updateReleaseInfo(releaseInfo));
-        releaseInfo.checks.map((check: CheckInfo) => {
-          return checkStatus(check.url).then(result => {
-            // Detect if the result changed, and notify!
-            const prevResult = prevResults[check.title];
-            if (prevResult && prevResult.status !== result.status) {
-              notifyChanges(check.title, result.status);
-            }
-            dispatch(addCheckResult(check.title, result));
-          });
-        });
-      })
-      .catch((err: string) => {
-        console.error(
-          `Failed getting the release info for ${versionToCheck}`,
-          err,
-        );
+    try {
+      const releaseInfo = await getReleaseInfo(versionToCheck);
+      dispatch(updateReleaseInfo(releaseInfo));
+      releaseInfo.checks.map(async (check: CheckInfo) => {
+        const result = await checkStatus(check.url);
+        const prevResult = prevResults[check.title];
+        if (prevResult && prevResult.status !== result.status) {
+          notifyChanges(check.title, result.status);
+        }
+        dispatch(addCheckResult(check.title, result));
       });
+    } catch (err) {
```

Comment:
we can't type this `err` (flow doesn't support it, see ), which is one drawback of using async/await instead of bare promises